File: backend/core/src/main/java/com/arg/swu/dto/NewsInfoData.java
package com.arg.swu.dto;

import com.arg.swu.models.commons.PageableCommon;

import io.swagger.v3.oas.annotations.media.Schema;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import lombok.Data;

/**
 *
 * @author kumpeep
 */
@Data
public class NewsInfoData extends PageableCommon implements Serializable {
	
    private static final long serialVersionUID = -8098525584987352173L;
    
	private Long newsId;
    private String newsHeading;
    private Long newsFormat;
    private String newsFormatNameEn;
    private String newsFormatNameTh;
    private String prefixShortTh;
    private String prefixShortEn;
    private String firstnameTh;
    private String firstnameEn;
    private String middlenameTh;
    private String middlenameEn;
    private String lastnameTh;
    private String lastnameEn;
    private Date newsStart;
    private Date newsEnd;
    private Date createDate;
    private Long createById;
    private String createdByNameTh;
    private String createdByNameEn;
    private Boolean newsStatus;
    private Long coursepublicRefId;
    private Long coursepublicRefIf;
    private String newsCoverimage;
    private String newsDetail;
    private Long newsOrder;
    private Long newsHilight;
    private Boolean newFlag;
    private Date newIconStart;
    private Date newIconEnd;
    
    private Boolean activeFlag;
    private Long newsAttachId;
    private String filename;
    private String prefix;
    private Long module;
    private String base64;
    
    @Schema(description = "custom เอกสารแนบ")
    private List<NewsInfoAttachData> newsAttachDocList;
    
}

File: frontend/src/app/models/news-management/NewsInfoData.ts
import { NewsInfoAttachData } from "./NewsInfoAttachData";

export interface NewsInfoData {
    newsId?:number;
    newsHeading?:string;
    newsFormat?:number;
    newsHilight?:number;
    newsFormatNameEn?:string;
    newsFormatNameTh?:string;
    prefixShortTh?:string;
    prefixShortEn?:string;
    firstnameTh?:string;
    firstnameEn?:string;
    middlenameTh?:string;
    middlenameEn?:string;
    lastnameTh?:string;
    lastnameEn?:string;
    newsStart?:Date;
    newFlag?:boolean;
    newIconStart?: Date;
    newIconEnd?:Date;
    newsEnd?:Date;
    newsCoverimage?:string;
    coursepublicRefId?:number;
    newsDetail?:string;
    newsOrder?:number;
    createById?:number;
    createDate?:Date;
    createdByName?:string;
    newsStatus?:boolean;
    rows?:number;
    first?: number;
    size?: number;
    filename?: string;
    prefix?: string;
    module?: number;
    mode?: string;
    activeFlag?: boolean;

    newsAttachDocList?: NewsInfoAttachData[];
}
